tap_surveymonkey/mode.py

```diff
@@ -157,12 +160,16 @@ def sync_survey_details(config, state):
     state['bookmarks'][stream_id] = {} if not state['bookmarks'].get(
         stream_id) else state['bookmarks'][stream_id]
     state['bookmarks'][stream_id]['full_sync'] = singer.utils.strftime(max_time)
+    singer.write_state(state)
 
     return state
 
 
 def sync_responses(config, state, simplify=False):
-    survey_id = config['survey_id']
+    survey_id = config.get('survey_id')
+    if not survey_id:
+        LOGGER.critical("survey id not provided. ")
```

Comment:
Could this be something like "Survey ID not provided. Syncing Responses requires a Survey ID"?

Reply:
Fixed as suggested.